BrainPortal/app/models/bourreau.rb

```diff
@@ -406,6 +406,13 @@ def self.process_command_start_workers(command)
     # Returns a logger object or the symbol :auto
     logger = self.initialize_worker_logger(log_to,verbose)
 
+    # prefetch git commits info just before workers are started (reduces git invocation #)
```

Comment:
This is still performed at the wrong place in the life cycle of the bourreau.

You are fetching the git revision numbers DURING the processing of the start workers command on the controls channel, this is not the role of the command interface. It doesn't know or shouldn't know anything about these specifics. This has lots of negative implications:

1. It would block the processing of the command until all the git revision info are received. On a slow filesystem (e.g. the home directory of a cluster), it would mean the start workers command would time out on the client side (suck commands are assumed to run in less than 10 seconds).
2. we'll be running this fetching code multiple times, every time we stop and start the workers, which is unnecessary

Move the code to a Bourreau system check instead

Reply:
Code is moved to the system check. I pushed new branch to reduce commit numbers, as it is same code being pushed around